api/breaker/round_tripper.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2022 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package breaker
+
+import (
+	"net/http"
+)
+
+// RoundTripper wraps a http.RoundTripper with a CircuitBreaker
+type RoundTripper struct {
+	tripper http.RoundTripper
+	cb      *CircuitBreaker
+}
+
+// NewRoundTripper returns a RoundTripper
+func NewRoundTripper(cb *CircuitBreaker, tripper http.RoundTripper) *RoundTripper {
+	return &RoundTripper{
+		tripper: tripper,
+		cb:      cb,
+	}
+}
+
+// RoundTrip forwards the request on to the provided http.RoundTripper if
+// the CircuitBreaker allows it
+func (t *RoundTripper) RoundTrip(request *http.Request) (*http.Response, error) {
+	v, err := t.cb.Execute(func() (interface{}, error) {
+		return t.tripper.RoundTrip(request)
+	})
+
+	if v != nil {
+		if res, ok := v.(*http.Response); ok {
```

Comment:
Do we really need to make this typesafe? And if we're so defensive against the behavior of `Execute`, shouldn't we also add a check to avoid returning `nil, nil` here?